web_sdk/BUILD.gn

```diff
@@ -43,60 +32,100 @@ group("web_sdk") {
   ]
 }
 
-prebuilt_dart_action("web_ui_sources") {
-  inputs = web_ui_sources + [ "sdk_rewriter.dart" ]
+template("sdk_rewriter") {
+  ui = defined(invoker.ui) && invoker.ui
+  if (!ui) {
+    assert(defined(invoker.library_name), "Must define 'library_name'")
+    assert(defined(invoker.api_file), "Must define 'api_file'")
+  }
+  assert(defined(invoker.input_dir), "Must define 'input_dir'")
+  assert(defined(invoker.output_dir), "Must define 'output_dir'")
```

Comment:
nit: maybe "pass" instead of "define"? These are just arguments passed to the template, no?